packages/neuron-wallet/src/services/indexer/queue.ts

```diff
@@ -43,6 +44,7 @@ export default class IndexerQueue {
   private resetFlag = false
 
   private latestCreatedBy: TxUniqueFlagCache = new TxUniqueFlagCache(100)
+  private txCache: TransactionCache = new TransactionCache(100)
```

Comment:
I don't see many benefits to introduce a specific cache implementation. Is using a `Map` directly not sufficient?

When a `limit` is set and the cache reaches that limit, pushing a new element always trigger the `find first and delete that one` action. Why is that even necessary.  (Another action to consider - just an example - when the limit is reached just clear the first 30 or so elements)

But my real argument here is we can just use a `Map` directly - not very necessary to have a thin layer wrapper of that.

Reply:
BTW, do we have a before and after time comparison for the Elon test wallet?

Reply:
One run on my machine. Before v.s. after: `50min` v.s. `25min`. Not bad.

Reply:
Not tested on the whole sync process, but I found `getTransaction` always need about `50ms` usually, if cached, it would take about `0~1ms` usually.
The loop would be more faster.